### PR TITLE
add: example script to kick the "Survivor" nickname

### PR DESCRIPTION
**Pull request by Draowyr**
- Added the "kicksurvivor.py" file, which prevents players with the default nickname "Survivor" from logging in.

### CODE REVIEW

File: examples/kicksurvivor.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""Connects to an RCON server and prevents players with the nickname 'Survivor' from logging in"""
+import re, math
+import asyncio
```
Comment (thegamecracks) [inline examples/kicksurvivor.py]:
```suggestion
"""Connects to an RCON server and kicks players with "Survivor" as their name."""
import asyncio
import math
import re
```

File: examples/kicksurvivor.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        player_name = match.group(2)
+        if player_name == "Survivor" or re.match(r'^Survivor \(\d+\)$', player_name):
+            await client.kick(int(player_id), kickReason)
+
+async def main():
+    async with client.connect(IP_ADDR, PORT, PASSWORD):
+        await asyncio.sleep(math.inf)
+
+if __name__ == "__main__":
+    asyncio.run(main())
```
Comment (thegamecracks) [inline examples/kicksurvivor.py]:
```suggestion
async def main():
    async with client.connect(IP_ADDR, PORT, PASSWORD):
        await asyncio.sleep(math.inf)


if __name__ == "__main__":
    asyncio.run(main())

```

File: examples/kicksurvivor.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+client = rcon.AsyncRCONClient()
+
+@client.dispatch.on_message
+async def on_message(message: str):
+    match = re.match(r'Player #(\d+) (\w+(?: \(\d+\))?) \(\d+\.\d+\.\d+\.\d+:\d+\) connected', message)
+    if match:
+        player_id = match.group(1)
+        player_name = match.group(2)
+        if player_name == "Survivor" or re.match(r'^Survivor \(\d+\)$', player_name):
+            await client.kick(int(player_id), kickReason)
```
Comment (thegamecracks) [inline examples/kicksurvivor.py]:
```suggestion
client = rcon.AsyncRCONClient()


@client.dispatch.on_player_connect
async def on_player_connect(player: rcon.Player):
    # Kick players with a name like "Survivor" or "Survivor (1)"
    if re.match(r"Survivor(?: \(\d+\))?", player.name) is not None:
        await player.kick("Name 'Survivor' not allowed")

```

**Review by thegamecracks — CHANGES_REQUESTED** [review]
Thanks for making a PR! Could you make the following changes before I accept it?

1. Rewrite the script using the [on_player_connect](https://berconpy.readthedocs.io/en/stable/generated/berconpy.AsyncEventDispatcher.html#berconpy.AsyncEventDispatcher.on_player_connect) event
2. Reformat the file with [Black](https://black.readthedocs.io/en/stable/index.html)
3. Rename the file to `kick_survivor.py`

The changes I've suggested will handle the first two steps, so you can accept those and rename the file afterwards.

**Comment by Draowyr** [comment]
I've made some changes to the file without using the [on_player_connect](https://berconpy.readthedocs.io/en/stable/generated/berconpy.AsyncEventDispatcher.html#berconpy.AsyncEventDispatcher.on_player_connect) event, which doesn't work with your suggested code. I've kept the fitre that lets you exclude players with the nickname "Survivor (1)" because BattlEye renames them when there are several connected.

**Comment by thegamecracks** [comment]
> I've made some changes to the file without using the [on_player_connect](https://berconpy.readthedocs.io/en/stable/generated/berconpy.AsyncEventDispatcher.html#berconpy.AsyncEventDispatcher.on_player_connect) event, which doesn't work with your suggested code.

I can only test this on Arma 3, but I had no issues with on_player_connect:

Survivor kicked

Your player regex is also equivalent to the one [used by the library](https://github.com/thegamecracks/berconpy/blob/v2.1.3/src/berconpy/parser.py#L50) internally. If on_player_connect doesn't work for you, I think that would count as a bug. Does the on_player_connect event fire at all? Does the player's name print out what you would expect? If the event isn't firing, could you send a snippet of the messages received by your client like:

```cpp
RCon admin #0 (127.0.0.1:60632) logged in
Player #0 Survivor (172.31.208.1:2316) connected
Player #0 Survivor (-) has been kicked by BattlEye: Admin Kick (Name 'Survivor' not allowed)
```

(redact any GUIDs or IP addresses as necessary)

---

> I've kept the fitre that lets you exclude players with the nickname "Survivor (1)" because BattlEye renames them when there are several connected.

Yup, I think that's fine to keep. I think you should rewrite it as a single regex pattern though:

```py
# Kick players with a name like "Survivor" or "Survivor (1)"
if re.match(r"Survivor(?: \(\d+\))?", player.name) is not None:
    await player.kick("Name 'Survivor' not allowed")
```

---

For styling, could you sort the imports and separate them into individual lines? I prefer that my codebase follows [PEP 8](https://peps.python.org/pep-0008/#imports) for style consistency. (in fact I probably should have a contributing guide that mentions this)

```py
import asyncio
import math
import re
```

Alternatively if you're fine with me making commits to your branch, I can update the styling myself :)

**Pull request merged**